### PR TITLE
update(docs): new readme style to introduce rules core concepts

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind feature

> /kind bug

> /kind cleanup

> /kind design

/kind documentation

> /kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area rules

> /area registry

> /area build

/area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

> /area maturity-stable

> /area maturity-incubating

> /area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:

Trying a different style to convey rules core concepts better. The new (visually more appealing) layout reduces some prior redundancy and contains extra valuable information. Hopefully the README is now more concise, relevant and easier to follow.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md
Comment (incertum) [inline README.md:47]: @falcosecurity/falco-maintainers @falcosecurity/rules-maintainers  is rules version 3.0.0 correct to ensure a clear transition to semver?

Comment (incertum) [reply README.md:47]: @LucaGuerra once we have the docs for "selective rules overrides" we can link to them as well.

File: README.md
Comment (incertum) [inline README.md:3]: Hopefully explicitly linking to the Supported Fields page may drive adoption of newer fields.

File: README.md
Comment (incertum) [inline README.md:32]: @falcosecurity/falco-maintainers @falcosecurity/rules-maintainers is rules version 3.0.0 correct to ensure a clear transition to semver?

@LucaGuerra once we have the docs for "selective rules overrides" we can link to them as well.

Comment (leogr) [reply README.md]: > @falcosecurity/falco-maintainers @falcosecurity/rules-maintainers is rules version 3.0.0 correct to ensure a clear transition to semver?

Ok for me!

cc @Andreagit97

**Review by leogr — APPROVED** [review]
LGMT

/hold

Please unhold close to the release

**Comment by incertum** [comment]
@leogr will you take care of unholding this once you feel we are close enough to the release?

**Comment by leogr** [comment]
> @leogr will you take care of unholding this once you feel we are close enough to the release?

Sure!

Also adding Luca in case I forget :angel: :sweat_smile: 
/assign @LucaGuerra

File: README.md
Comment (leogr) [inline README.md:3]: It would be nice to use https://github.com/falcosecurity/rules/blob/main/.github/FALCO_VERSIONS as input to generate the list of compatible versions.

Not a blocker for now. Just an idea.

Comment (incertum) [reply README.md:3]: Yes this would be very cool! Would suggest regarding this PR finished and perhaps @leogr and @LucaGuerra you could open a small follow up PR once we have the next Falco version released. That PR could then also include more links and cross-references to new docs we have yet to create.

**Comment by incertum** [comment]
Since concerns around missing fields keep coming up, thought we should add another item "Bugs and Debugging" to the readme, linking to our Install and Operate Guides that will include new Help Guides soon.

**Comment by incertum** [comment]
Now links to https://falco.org/docs/troubleshooting/ (needs to be synced with https://github.com/falcosecurity/falco-website/pull/1246).

**Review by leogr — APPROVED** [review]
LGTM

/hold for 0.37 and docs update

**Comment by leogr** [comment]
> Now links to https://falco.org/docs/troubleshooting/ (needs to be synced with [falcosecurity/falco-website#1246](https://github.com/falcosecurity/falco-website/pull/1246)).

[troubleshooting/](https://falco.org/docs/troubleshooting/) is live now.

/hold cancel

**Pull request merged**
